### DISCUSSION TITLE
如图，想让红色方块在线条居中，请问该怎么设置？

### DISCUSSION THREAD

**Discussion by PineSongCN** [antvis/X6#4433] [Q&A]
![image](https://github.com/user-attachments/assets/82e0cffc-ec46-484e-9624-ef247d505136)

**Comment by resetsix** [comment]
```tsx
{
  markup: [
    {
      tagName: 'rect',
      selector: 'body',
    },
    {
      tagName: 'text',
      selector: 'label',
    },
  ],
  attrs: {
    text: {
      fill: '#000',
      fontSize: 14,
      textAnchor: 'middle',
      textVerticalAnchor: 'middle',
      pointerEvents: 'none',
    },
    rect: {
      ref: 'label',
      fill: '#fff',
      rx: 3,
      ry: 3,
      refWidth: 1,
      refHeight: 1,
      refX: 0,
      refY: 0,
    },
  },
  // 修改这里
  position: {
    distance: 0.5 // default 0.5 表示居中位置
  },
}
```
- [参考文档](https://x6.antv.antgroup.com/api/model/labels#%E4%BD%8D%E7%BD%AE)
- [参考示例](https://x6.antv.antgroup.com/zh/examples/edge/label/#multi-labels)

可以将「示例」中的`distance: 0.3`值修改为`distance: 0.5`，即使存在多个 label 也没问题。

**Reply by PineSongCN** [reply]
> * 参考示例

感谢，红色方块其实都是节点，不是标签

**Reply by resetsix** [reply]
> > * 参考示例
> 
> 感谢，红色方块其实都是节点，不是标签

无论是标签还是节点都是内容在 edge 上的位置，无所谓的。

**Reply by PineSongCN** [reply]
> ```tsx
> 
>   position: {
>     distance: 0.5 // default 0.5 表示居中位置
>   },
> ```

嗯嗯，因为是节点，确切地说是要控制边分叉的拐点，试了改变节点x的位置，但是节点总是如图的偏左。
